### PR TITLE
[plugin] OOT support and abstraction for flash attention backend

### PR DESCRIPTION
**Pull request by DevashishLal-CB**
RFC: https://github.com/sgl-project/sglang/issues/26426

## Motivation

This change introduces the MultiPlatformDispatcher which is similar to the MultiPLatformOp but not a torch.nn.Module, the dispatcher enables abstracting the attention kernel calls in the flash attention backend allowing OOT plugins to re-use the backend as long as they provide kernels that implement the flash attention api

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :no_entry_sign: [Run #26476434189](https://github.com/sgl-project/sglang/actions/runs/26476434189)
Latest PR Test (Extra): :x: [Run #26476433998](https://github.com/sgl-project/sglang/actions/runs/26476433998)

### CODE REVIEW

File: python/sglang/srt/layers/attention/attention_registry.py [hunk 1]
```diff
@@ -171,6 +172,12 @@ def create_flashmla_backend(runner):
 
 @register_attention_backend("fa3")
 def create_flashattention_v3_backend(runner):
+    if current_platform.is_out_of_tree():
```
Comment (BBuf) [inline python/sglang/srt/layers/attention/attention_registry.py:175]: Can we add a small OOT-dispatch test or an early validation here? In the OOT branch we return `FlashAttentionBackend` unconditionally, but if a platform has not registered handlers for `FlashAttnVarlenFuncOp`, `FlashAttnWithKvcacheOp`, and `MergeStateV2Op`, dispatch falls back to `forward_native`, which imports `sgl_kernel`. That makes the OOT path fail late with a CUDA/sgl_kernel error instead of a clear plugin-contract error.

File: python/sglang/srt/layers/attention/flashattention_backend.py [hunk 2]
```diff
@@ -25,12 +26,236 @@
...
+    def _resolve_impl(self):
+        if self.fa_impl_ver == 3:
+            from sgl_kernel.flash_attn import flash_attn_varlen_func
+        elif self.fa_impl_ver == 4:
+            from sglang.jit_kernel.flash_attention_v4 import flash_attn_varlen_func
+        else:
+            raise ValueError(f"Invalid fa_impl_ver: {self.fa_impl_ver}")
+        return flash_attn_varlen_func
+
+    def forward_native(
```
Comment (BBuf) [inline python/sglang/srt/layers/attention/flashattention_backend.py:47]: These wrappers now define the effective kernel call surface for the backend. They currently mirror only the kwargs used by this file, while the underlying flash-attn APIs support additional args such as `score_mod`, `aux_tensors`, `attention_chunk`, `pack_gqa`, and descales. Could we either keep a `**kwargs` passthrough / match the underlying signatures, or document this as the SRT FA-backend subset? That would reduce churn when new FA kwargs are threaded through.

File: python/sglang/srt/layers/attention/attention_registry.py (hunk 1 above)
Comment (DevashishLal-CB) [reply python/sglang/srt/layers/attention/attention_registry.py:175]: I agree, I am still working on this PR will mark it as draft till it's ready

File: python/sglang/srt/layers/attention/flashattention_backend.py (hunk 2 above)
Comment (DevashishLal-CB) [reply python/sglang/srt/layers/attention/flashattention_backend.py:47]: yeah, will add the passthrough

File: python/sglang/srt/layers/attention/flashattention_backend.py [hunk 3]
```diff
@@ -3307,5 +3542,5 @@ def cdiv(a: int, b: int) -> int:
 
 # TODO(hebiao064): remove this once we have a better way to handle the merge_state_v2 torch.compile issue
 @torch._dynamo.disable()
-def merge_state_v2_wrapper(o, s_a, o_exp, s_b):
-    return merge_state_v2(o, s_a, o_exp, s_b)
+def merge_state_v2_wrapper(merge_op, o, s_a, o_exp, s_b):
```
Comment (AgainstEntropy) [inline python/sglang/srt/layers/attention/flashattention_backend.py:3545]: `merge_state_v2_wrapper` is also imported and used by musa and xpu.
We need to make sure the function signature change doesn't break the original behavior.

File: python/sglang/srt/layers/utils/multi_platform.py [hunk 4]
```diff
@@ -132,3 +82,63 @@ def dispatch_forward(self):
             return self.forward_musa
         else:
             return self.forward_native
+
+
+class MultiPlatformDispatcher(_MultiPlatformBase):
+    def __init__(self):
+        self._forward_method: Callable = self.dispatch_forward()
+
+    def __call__(self, *args, **kwargs):
```
Comment (AgainstEntropy) [inline python/sglang/srt/layers/utils/multi_platform.py:91]: Seems we can keep the `@debug_kernel_api` decorator? Do you have any comments here? @BBuf
